bftengine/src/bcstatetransfer/SourceSelector.hpp

```diff
@@ -134,6 +144,17 @@ class SourceSelector {
   uint16_t nominatedPrimaryCounter_ = 0;
   uint16_t minPrePrepareMsgsForPrimaryAwarness_ = 10;
   logging::Logger &logger_;
+
+ public:
+  // Metric
+  concordMetrics::Component metrics_component_;
+  struct Metrics {
```

Comment:
should be protected and manage it's own metrics logic, as said before. No metrics should be changed from outside.
Add required API to support change triggers and printouts to logs or to communication ports.
If you think the added part of the refactoring is too much, handle only the replacement metrics for now, but we should open a new task for that.

Reply:
Thanks. Addressed.